### ISSUE TITLE
Implementing differential privacy for the bounded-norm vector type

### ISSUE THREAD

**Issue by MxmUrw** [divviup/libprio-rs#440]
We are now at the point where we would like to add differential privacy to our [bounded-norm fixed-point vector type](https://github.com/divviup/libprio-rs/pull/283). As was pointed out to us, there has been some discussion about the best way to integrate dp with VDAF / DAP [here](https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/94) and [here](https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/19).

Have there been any further developments with regards to dp since those two issues?

It looks like the simplest way to achieve our version of DP (each of the aggregators adds a pre-configured amount of noise to the aggregate share) would be by doing the following:
 - Extend the Prio3 `Type` trait to require an additional function `add_noise(&self, aggregate_share: Vec<Field>) -> Vec<Field>`.
 - Call this function in the `Aggregator::aggregate()` implementation for `Prio3` on the resulting aggregate share.
 - In the implementation of `add_noise` for our vector type, do add the (discrete gaussian) noise as required by our encoding.
 - Add a default implementation for `add_noise()` that is the identity function, so other prio types do not need to be changed.

This approach seems to be the most straight-forward to us, but it possibly disregards the proposed solutions in the above-mentioned threads. We would like to discuss about how best to align our efforts with the future specification of dp in VDAF (if such a thing is indeed going to be specified).

**Comment by cjpatton** [comment]
I believe this can be closed.

**Comment by tgeoghegan** [comment]
I agree. Work landed here back in 2023, and we can always reopen or open new issues for future work.
